src/CoherenceBuild/CoherenceBuild.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+
+namespace CoherenceBuild
+{
+    public class CoherenceBuild
+    {
+        private static readonly object _processPackageLock = new object();
+        private List<RepoInfo> _reposToProcess;
+        private readonly string _dropFolder;
+        private readonly string _buildBranch;
+        private readonly string _outputPath;
+        private readonly string _nugetPublishFeed;
+        private readonly string _apiKey;
+
+        public CoherenceBuild(
+            List<RepoInfo> reposToProcess,
+            string dropFolder,
+            string buildBranch,
+            string outputPath,
+            string nugetPublishFeed,
+            string apiKey)
+        {
+            _reposToProcess = reposToProcess;
+            _dropFolder = dropFolder;
+            _buildBranch = buildBranch;
+            _outputPath = outputPath;
+            _nugetPublishFeed = nugetPublishFeed;
+            _apiKey = apiKey;
+        }
+
+        public bool VerifyPackages { get; set; } = true;
+
+        public bool VerifyPartnerPackages { get; set; } = true;
+
+        public int Execute()
+        {
+            if (!Directory.Exists(_dropFolder))
+            {
+                Log.WriteError($"Drop share {_dropFolder} does not exist");
+                return 1;
```

Comment:
We do it elsewhere, and I've gotten this feedback in other reviews, please do :)